src/App.tsx

```diff
@@ -1,14 +1,63 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { getTodos } from './api';
+import { Option } from './types/Option';
+import { Todo } from './types/Todo';
+
+const filteredTodos = (
+  todos: Todo[],
+  query: string,
+  option: string,
+) => {
+  let filteredByOption: Todo[] = [];
+
+  if (option) {
+    filteredByOption = todos.filter((todo) => {
+      switch (option) {
+        case Option.ACTIVE:
+          return !todo.completed;
+
+        case Option.COMPLETED:
+          return todo.completed;
+
+        default:
+          return todo;
+      }
+    });
+  }
+
+  if (query) {
+    filteredByOption = filteredByOption.filter(todo => (
+      todo.title.toLowerCase().includes(query.trim().toLowerCase())
+    ));
+  }
+
+  return filteredByOption;
+};
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [loading, setLoading] = useState(false);
```

Comment:
```suggestion
  const [isLoading, setIsLoading] = useState(false);
```
![image](https://github.com/mate-academy/react_dynamic-list-of-todos/assets/96409563/5209f624-4ae1-449f-bc61-c4bf26420a62)